interface.go

```diff
@@ -20,3 +21,34 @@ type Session interface {
 	RemoteAddr() net.Addr
 	Close(error) error
 }
+
+// ConnState is the status of the connection
+type ConnState int
+
+const (
+	// ConnStateVersionNegotiated means that version negotiation is complete
+	ConnStateVersionNegotiated ConnState = iota
+	// ConnStateSecure means that the connection is encrypted
+	ConnStateSecure
+	// ConnStateForwardSecure means that the connection is forward secure
+	ConnStateForwardSecure
```

Comment:
Server:

There's a period of time between ConnStateVersionNegotiated and ConnStateSecure, e.g. if the client has to present STKs before the server sends it the full SCFG. During this phase, the server sends insecure data.

I agree that for the server there's no period between receiving the first (fw) secure data and sending the first (fw) secure data.

Client:

I don't think ConnStateSecure should fire twice. Our API hides the effects of 0-RTT (we don't need the user to re-`Write` stream data after all), so I don't see why we should do this here.

Still, there's a period of time during a slow handshake where we are sending (fw) secure data but receiving insecure data. Are you saying that doesn't convey any information that warrants escalating through our API?